### ISSUE TITLE
CLI does not work behind proxy

### ISSUE THREAD

**Issue by minijus** [crowdin/crowdin-cli#306]
**Describe the bug**
Configuring HTTP_PROXY_* environment variables does not affect CLI behaviour. It is not possible to use CLI behind proxy.

**To Reproduce**
1. Run CLI with proxy configured. Based on code here https://github.com/crowdin/crowdin-cli/blob/89e4684cdfad9a61ad86616255ca32551f3a8761/src/main/java/com/crowdin/cli/Cli.java#L31
```bash
HTTP_PROXY_HOST=proxy.example.com HTTP_PROXY_PORT=8080 HTTP_PROXY_USER=user HTTP_PROXY_PASSWORD=password java -jar crowdin-cli.jar upload
```
2. Get an error
```
[ERROR] Fetching project info
java.lang.RuntimeException: Failed to collect project info. Please contact our support team for help
        at com.crowdin.cli.utils.console.ConsoleSpinner.execute(ConsoleSpinner.java:29)
        at com.crowdin.cli.commands.actions.UploadSourcesAction.act(UploadSourcesAction.java:65)
        at com.crowdin.cli.commands.picocli.ClientActCommand.act(ClientActCommand.java:25)
        at com.crowdin.cli.commands.picocli.GenericCommand.run(GenericCommand.java:57)
        at picocli.CommandLine.executeUserObject(CommandLine.java:1919)
        at picocli.CommandLine.access$1100(CommandLine.java:145)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2332)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2326)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2291)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2159)
        at picocli.CommandLine.execute(CommandLine.java:2058)
        at com.crowdin.cli.commands.picocli.PicocliRunner.execute(PicocliRunner.java:36)
        at com.crowdin.cli.Cli.main(Cli.java:19)
Caused by: java.lang.RuntimeException: Error from server: <Code: <empty_code>, Message: Connect to api.crowdin.com:443 [api.crowdin.com/34.233.13.83, api.crowdin.com/3.220.221.31, api.crowdin.com/107.21.48.25] failed: Connection timed out: connect>
        at com.crowdin.cli.client.CrowdinClientCore.executeRequest(CrowdinClientCore.java:109)
        at com.crowdin.cli.client.CrowdinClientCore.executeRequest(CrowdinClientCore.java:82)
        at com.crowdin.cli.client.CrowdinClient.getProject(CrowdinClient.java:97)
        at com.crowdin.cli.client.CrowdinClient.populateProjectWithInfo(CrowdinClient.java:81)
        at com.crowdin.cli.client.CrowdinClient.downloadFullProject(CrowdinClient.java:42)
        at com.crowdin.cli.utils.console.ConsoleSpinner.execute(ConsoleSpinner.java:24)
        ... 12 more
```

**Expected behavior**
It would be great to have clear documentation how to use CLI with proxy configuration.

Works fine:
```bash
export https_proxy=http://user:password@proxy.example.com
curl https://api.crowdin.com
```

**Comment by minijus** [comment]
Update: I did the regression with version 2 and proxy configuration worked fine.

**Comment by nataliacrowdin** [comment]
Hi there!
Thank you for bringing this to our attention
Already checking everything with our developers

**Comment by andrii-bodnar** [comment]
Hi @minijus !

There is a pre-release build of CLI with the fix (not yet deployed) [crowdin-cli-3.2.2.zip](https://github.com/crowdin/crowdin-cli/files/5169813/crowdin-cli-3.2.2.zip)

Could you please try this build to check if everything is ok?

**Comment by minijus** [comment]
> Hi @minijus !
> 
> There is a pre-release build of CLI with the fix (not yet deployed) [crowdin-cli-3.2.2.zip](https://github.com/crowdin/crowdin-cli/files/5169813/crowdin-cli-3.2.2.zip)
> 
> Could you please try this build to check if everything is ok?

Hi!

the fix on new build worked like a charm :)

**Comment by minijus** [comment]
@andrii-bodnar some update, the version with fix worked fine for upload, but not for download.
I get this exception:
```
10:33:23  java.lang.RuntimeException: Failed to write to the file '%s'. Try to run the application with admin permission
10:33:23  	at com.crowdin.cli.commands.actions.DownloadAction.downloadTranslations(DownloadAction.java:343)
10:33:23  	at com.crowdin.cli.commands.actions.DownloadAction.act(DownloadAction.java:143)
10:33:23  	at com.crowdin.cli.commands.picocli.ClientActCommand.act(ClientActCommand.java:25)
10:33:23  	at com.crowdin.cli.commands.picocli.GenericCommand.run(GenericCommand.java:57)
10:33:23  	at picocli.CommandLine.executeUserObject(CommandLine.java:1919)
10:33:23  	at picocli.CommandLine.access$1100(CommandLine.java:145)
10:33:23  	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2332)
10:33:23  	at picocli.CommandLine$RunLast.handle(CommandLine.java:2326)
10:33:23  	at picocli.CommandLine$RunLast.handle(CommandLine.java:2291)
10:33:23  	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2159)
10:33:23  	at picocli.CommandLine.execute(CommandLine.java:2058)
10:33:23  	at com.crowdin.cli.commands.picocli.PicocliRunner.execute(PicocliRunner.java:36)
10:33:23  	at com.crowdin.cli.Cli.main(Cli.java:14)
10:33:23  Caused by: java.net.UnknownHostException: crowdin-packages.downloads.crowdin.com
10:33:23  	at java.base/java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:220)
10:33:23  	at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:403)
10:33:23  	at java.base/java.net.Socket.connect(Socket.java:609)
10:33:23  	at java.base/sun.security.ssl.SSLSocketImpl.connect(SSLSocketImpl.java:289)
10:33:23  	at java.base/sun.security.ssl.BaseSSLSocketImpl.connect(BaseSSLSocketImpl.java:173)
10:33:23  	at java.base/sun.net.NetworkClient.doConnect(NetworkClient.java:182)
10:33:23  	at java.base/sun.net.www.http.HttpClient.openServer(HttpClient.java:474)
10:33:23  	at java.base/sun.net.www.http.HttpClient.openServer(HttpClient.java:569)
10:33:23  	at java.base/sun.net.www.protocol.https.HttpsClient.<init>(HttpsClient.java:265)
10:33:23  	at java.base/sun.net.www.protocol.https.HttpsClient.New(HttpsClient.java:372)
10:33:23  	at java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.getNewHttpClient(AbstractDelegateHttpsURLConnection.java:191)
10:33:23  	at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1187)
10:33:23  	at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1081)
10:33:23  	at java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:177)
10:33:23  	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1592)
10:33:23  	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1520)
10:33:23  	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:250)
10:33:23  	at java.base/java.net.URL.openStream(URL.java:1140)
10:33:23  	at com.crowdin.cli.commands.actions.DownloadAction.downloadTranslations(DownloadAction.java:340)
10:33:23  	... 12 more
```

I am quite sure the permissions are correct. What caught my eye was `Caused by: java.net.UnknownHostException: crowdin-packages.downloads.crowdin.com`.

**Comment by andrii-bodnar** [comment]
Hi @minijus ! Sorry for the inconvenience, we will check this

**Comment by minijus** [comment]
Looked quickly into code and noticed that for translations download `url.openStream()` is used instead of client instance.
https://github.com/crowdin/crowdin-cli/blob/89e4684cdfad9a61ad86616255ca32551f3a8761/src/main/java/com/crowdin/cli/commands/actions/DownloadAction.java#L340

Thus, when passing https proxy flags directly to java it worked fine:
```bash
HTTP_PROXY_HOST=proxy.example.com \
HTTP_PROXY_PORT=8080 \
HTTP_PROXY_USER=user \
HTTP_PROXY_PASSWORD=password \
CROWDIN_API_KEY=key \
CROWDIN_BASE_PATH=${pwd} \
java \
	-Dhttps.proxyHost=proxy.example.com  \
	-Dhttps.proxyPort=8080 \
	-Dhttps.proxyUser=user \
	-Dhttps.proxyPassword=password \
	-jar crowdin-cli-3.2.2.jar download 
```

**Comment by andrii-bodnar** [comment]
Hi @minijus !

There is another pre-release build which includes proxy translations download fix - [crowdin-cli-3.2.2.zip](https://github.com/crowdin/crowdin-cli/files/5208649/crowdin-cli-3.2.2.zip)

P.S. This build is not completely tested by our QA team yet.

**Comment by minijus** [comment]
> Hi @minijus !
> 
> There is another pre-release build which includes proxy translations download fix - [crowdin-cli-3.2.2.zip](https://github.com/crowdin/crowdin-cli/files/5208649/crowdin-cli-3.2.2.zip)
> 
> P.S. This build is not completely tested by our QA team yet.

Hi!

Thanks for this update. I can confirm that this build works fine (both upload and download) and there is no need to pass extra proxy properties directly to java.
